Sources/BraveWallet/Crypto/Asset Details/NFTDetailView.swift

```diff
@@ -91,30 +91,55 @@ struct NFTDetailView: View {
-                    nftURL = URL(string: "\(baseURL)?a=\(tokenId)")
+            if nftDetailStore.nft.isErc721 {
+              HStack {
+                Text(Strings.Wallet.nftDetailTokenID)
+                  .font(.headline.weight(.semibold))
+                Spacer()
+                Button(action: {
+                  if let explorerURL = nftDetailStore.networkInfo.blockExplorerUrls.first {
+                    let baseURL = "\(explorerURL)/token/\(nftDetailStore.nft.contractAddress)"
+                    var nftURL = URL(string: baseURL)
+                    if let tokenId = Int(nftDetailStore.nft.tokenId.removingHexPrefix, radix: 16) {
+                      nftURL = URL(string: "\(baseURL)?a=\(tokenId)")
+                    }
+                    
+                    if let url = nftURL {
+                      openWalletURL?(url)
+                    }
                   }
-                  
-                  if let url = nftURL {
-                    openWalletURL?(url)
+                }) {
+                  if let tokenId = Int(nftDetailStore.nft.tokenId.removingHexPrefix, radix: 16) {
+                    Text(verbatim: "#\(tokenId)")
+                      .foregroundColor(Color(.braveBlurple))
+                  } else {
+                    Text("\(nftDetailStore.nft.name) #\(nftDetailStore.nft.tokenId)")
+                      .foregroundColor(Color(.braveBlurple))
                   }
                 }
-              }) {
-                if let tokenId = Int(nftDetailStore.nft.tokenId.removingHexPrefix, radix: 16) {
-                  Text(verbatim: "#\(tokenId)")
-                    .foregroundColor(Color(.braveBlurple))
-                } else {
-                  Text("\(nftDetailStore.nft.name) #\(nftDetailStore.nft.tokenId)")
+              }
+            } else {
+              HStack {
+                Text(Strings.Wallet.tokenMintAddress)
+                  .font(.headline.weight(.semibold))
+                Spacer()
+                Button(action: {
+                  if WalletConstants.supportedTestNetworkChainIds.contains(nftDetailStore.networkInfo.chainId) {
+                    if let components = nftDetailStore.networkInfo.blockExplorerUrls.first?.separatedBy("/?cluster="), let baseURL = components.first {
+                      let cluster = components.last ?? ""
+                      if let nftURL = URL(string: "\(baseURL)/address/\(nftDetailStore.nft.contractAddress)/?cluster=\(cluster)") {
+                        openWalletURL?(nftURL)
+                      }
+                    }
+                  } else {
+                    if let explorerURL = nftDetailStore.networkInfo.blockExplorerUrls.first, let nftURL = URL(string: "\(explorerURL)/address/\(nftDetailStore.nft.contractAddress)") {
+                      openWalletURL?(nftURL)
+                    }
+                  }
+                }) {
+                  Text("\(nftDetailStore.nft.contractAddress.truncatedAddress)")
```

Comment:
<img width="294" alt="Screenshot 2023-01-16 at 2 08 37 PM" src="https://user-images.githubusercontent.com/5314553/212750759-9c63797e-ce66-491c-8856-0aeff08589e5.png">
I think we are missing the 'open' icon here? Looks like this was missed when we added ERC721

Reply:
fixed in [8b343eb](https://github.com/brave/brave-ios/pull/6771/commits/8b343eb9ec23f03eb6a03154ce1c857c347bb0a6)